### PR TITLE
feat: able to keep Worker's state into the boltdb.

### PR DESCRIPTION
**Pull request by sshaman1101**
Changed:
- UUID (Worker ID) is placed into new storage.
- Config param with path to boltdb file.
- UUIDpath config's parameter where removed.

### CODE REVIEW

File: insonmnia/miner/state.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	ctx context.Context
+	s   store.Store
+
+	uuid string
+}
+
+func initStorage(p string) (store.Store, error) {
+	boltdb.Register()
+	config := store.Config{
+		Bucket: "sonm",
```
Comment (3Hren) [inline insonmnia/miner/state.go]: Make configurable, but still optional.

File: insonmnia/miner/state.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+func newEmptyState() *stateJSON {
+	return &stateJSON{
+		UUID: uuid.New(),
+	}
+}
+
+type state struct {
+	mu  sync.Mutex
+	ctx context.Context
+	s   store.Store
```
Comment (3Hren) [inline insonmnia/miner/state.go]: Come on. Give this a meaningful name.

File: insonmnia/miner/state.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+// loadInitial loads state from boltdb
+func (s *state) loadInitial() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	kv, err := s.s.Get(stateKey)
+	if err != nil && err != store.ErrKeyNotFound {
+		return err
+
```
Comment (3Hren) [inline insonmnia/miner/state.go]: Blank.

File: insonmnia/miner/state.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	b, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	return s.s.Put(stateKey, b, &store.WriteOptions{})
+}
+
+func (s *state) getID() string {
```
Comment (3Hren) [inline insonmnia/miner/state.go]: Just `ID` or `UUID`.

File: insonmnia/miner/state.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+func newEmptyState() *stateJSON {
+	return &stateJSON{
+		UUID: uuid.New(),
+	}
+}
+
+type state struct {
+	mu  sync.Mutex
```
Comment (3Hren) [inline insonmnia/miner/state.go]: Does this **really** require thread-safe access? Seems not.

**Comment by sshaman1101** [comment]
This PR is completely reworked by #530, in the current state it's completely useless.

**Pull request closed without merging**